lib/pure/nativesockets.nim

```diff
@@ -374,6 +374,19 @@ proc getHostByName*(name: string): Hostent {.tags: [ReadIOEffect].} =
   result.addrList = cstringArrayToSeq(s.h_addr_list)
   result.length = int(s.h_length)
 
+proc getHostname*(): string {.tags: [ReadIOEffect].} =
+  ## Returns the local hostname (not the FQDN)
+  const size = 64
```

Comment:
I should have put in a comment:
https://tools.ietf.org/html/rfc1035#section-2.3.1
https://tools.ietf.org/html/rfc2181#section-11
https://en.wikipedia.org/wiki/Hostname
https://technet.microsoft.com/en-us/library/cc959336.aspx